### PR TITLE
fix frame, generic, series.py

### LINKED ISSUE #38138: CLN: C408 Unnecessary dict call - rewrite as a literal
The following files contain flake8 C408 Unnecessary dict call - rewrite as a literal. Good for first time contributors. See #38078 and #38116 for how to fix them. Use `flake8 [file_path]` to check for each file.

### C408
- [x] ./pandas/core/generic.py
- [x] ./pandas/core/series.py
- [x] ./pandas/core/frame.py
- [x] ./pandas/core/strings/accessor.py
- [x] ./pandas/core/dtypes/dtypes.py
- [x] ./pandas/core/computation/pytables.py
- [x] ./pandas/core/computation/expressions.py
- [x] ./pandas/core/arrays/floating.py
- [x] ./pandas/core/arrays/interval.py
- [x] ./pandas/core/arrays/numpy_.py
- [x] ./pandas/core/arrays/base.py
- [x] ./pandas/core/ops/methods.py
- [x] ./pandas/core/indexes/interval.py
- [x] ./pandas/core/indexes/multi.py
- [x] ./pandas/core/indexes/numeric.py
- [x] ./pandas/core/indexes/base.py
- [x] ./pandas/io/pytables.py
- [x] ./pandas/tests/reshape/concat/test_datetimes.py
- [x] ./pandas/tests/reshape/merge/test_merge.py
- [x] ./pandas/tests/reshape/merge/test_multi.py
- [x] ./pandas/tests/tools/test_to_numeric.py
- [x] ./pandas/tests/resample/test_time_grouper.py
- [x] ./pandas/tests/util/test_assert_series_equal.py
- [x] ./pandas/tests/util/test_assert_interval_array_equal.py
- [x] ./pandas/tests/util/test_assert_index_equal.py
- [x] ./pandas/tests/util/test_assert_extension_array_equal.py
- [x] ./pandas/tests/io/test_parquet.py
- [x] ./pandas/tests/io/generate_legacy_storage_files.py
- [x] ./pandas/tests/io/formats/test_format.py
- [x] ./pandas/tests/io/parser/test_comment.py
- [x] ./pandas/tests/io/parser/test_parse_dates.py
- [x] ./pandas/tests/io/parser/test_c_parser_only.py
- [x] ./pandas/tests/io/parser/test_dialect.py
- [x] ./pandas/tests/io/parser/test_skiprows.py
- [x] ./pandas/tests/io/parser/test_encoding.py
- [x] ./pandas/tests/io/parser/test_read_fwf.py
- [x] ./pandas/tests/io/parser/test_python_parser_only.py
- [x] ./pandas/tests/io/parser/test_na_values.py
- [x] ./pandas/tests/io/parser/test_usecols.py
- [x] ./pandas/tests/io/parser/test_quoting.py
- [x] ./pandas/tests/io/parser/test_header.py
- [x] ./pandas/tests/io/parser/test_index_col.py
- [x] ./pandas/tests/io/json/test_ujson.py
- [x] ./pandas/tests/io/json/test_normalize.py
- [x] ./pandas/tests/io/pytables/test_timezones.py
- [x] ./pandas/tests/io/pytables/test_store.py
- [x] ./pandas/tests/frame/test_reductions.py
- [x] ./pandas/tests/frame/test_arithmetic.py
- [x] ./pandas/tests/frame/methods/test_to_records.py
- [x] ./pandas/tests/frame/methods/test_sort_values.py
- [x] ./pandas/tests/frame/methods/test_fillna.py
- [x] ./pandas/tests/frame/methods/test_to_csv.py
- [x] ./pandas/tests/frame/indexing/test_indexing.py
- [x] ./pandas/tests/frame/indexing/test_where.py
- [x] ./pandas/tests/dtypes/test_inference.py
- [x] ./pandas/tests/groupby/test_function.py
- [x] ./pandas/tests/groupby/transform/test_transform.py
- [x] ./pandas/tests/plotting/test_boxplot_method.py
- [x] ./pandas/tests/plotting/frame/test_frame_subplots.py
- [ ] ./pandas/tests/io/parser/test_common.py

### Remove ignore setting from flake8 section
- [ ] setup.cfg

### PR DESCRIPTION
**Pull request by liudj2008**
- [x] xref #38138
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry
        Identify unnecessary dict call - rewrite as a literal

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -8196,8 +8196,8 @@ def resample(
         For DataFrame objects, the keyword `on` can be used to specify the
         column instead of the index for resampling.
 
-        >>> d = dict({'price': [10, 11, 9, 13, 14, 18, 17, 19],
-        ...           'volume': [50, 60, 40, 100, 50, 100, 40, 50]})
+        >>> d = {'price': [10, 11, 9, 13, 14, 18, 17, 19],
+        ...      'volume': [50, 60, 40, 100, 50, 100, 40, 50]}
```
Comment (ivanovmg) [inline pandas/core/generic.py:8200]: Good catch here!

**Review by ivanovmg — APPROVED** [review]
@liudj2008, thank you for the PR!
Looks good to me.
Please update the original message: "partially closes" -> "xref".

**Comment by jreback** [comment]
thanks @liudj2008

**Comment by liudj2008** [comment]
Thank you @ivanovmg and @jreback . Will look more issues with pandas!

**Pull request merged**